### PR TITLE
Revisiting Rock Paper Scissors Assignment: revise Git instruction 

### LINKED ISSUE #25763: Foundations: Inconsistent Git instructions
### Describe your suggestion

Step 3 in the Revisiting Rock Paper Scissors assignment instructions currently says this:
> Once you’re all done with your UI and made sure everything’s satisfactory, commit your changes to the rps-ui branch.

This implies the student should go against the advice to commit early and often from the top of the assignment instructions and [the commit messages lesson](https://www.theodinproject.com/lessons/foundations-commit-messages), which I don't think is intended, right? It gave me a little pause when going through this lesson, and then I saw someone else ask for clarification about it on Discord as well.

I looked through the blame for why this step was written, and found this commit, explaining that the motivation was to make sure people knew to commit before merging. 59b9474 

So a change would have to both remove this implication while still keeping the value that commit added. Maybe something like this?
> Once you’re all done with your UI and made sure everything’s satisfactory, ensure all of your changes are committed to the rps-ui branch with `git status` before continuing.

### Path

Foundations

### Lesson Url

https://www.theodinproject.com/lessons/foundations-revisiting-rock-paper-scissors#assignment

### Checks

- [X] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
- [ ] Would you like to work on this issue?

### (Optional) Discord Name

_No response_

### (Optional) Additional Comments

Having a great time with TOP so far, thank you to everyone who's worked so hard on it!

### PR DESCRIPTION
**Pull request by CBID2**
## Because

Step 3 in the Revisiting Rock, Paper, Scissors assignment contradicts the instruction mentioned in the commit messages lesson, so a student made a suggestion on how to revise to solve this problem 

## This PR
 Makes the Git instruction consistent with the steps mentioned in the Commit messages lesson. 

## Issue

Closes #25763 

## Additional Information

This does not correlate to the issue, but I received a lot of Markdown errors in this lesson. Should I fix them?

## Pull Request Requirements

-   [X] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
-   [X] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`
-   [X] The `Because` section summarizes the reason for this PR
-   [X] The `This PR` section has a bullet point list describing the changes in this PR
-   [X] If this PR addresses an open issue, it is linked in the `Issue` section
-   [X] If any lesson files are included in this PR, they have been previewed with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure it is formatted correctly
-   [X] If any lesson files are included in this PR, they follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

### CODE REVIEW

**Comment by crystalsolenoid** [comment]
The content change looks in line with what I was envisioning in 25763. Thank you for doing the PR.

I am not a maintainer and am new to contributing to this repo, but I believe it's generally (unless there's a repo-specific rule not to) good git practice to not combine content changes and grammar fixes in the same commit.

In this case, it made it very difficult to find the relevant change in the diff. If this content change needed to be reverted for any reason, it would then also undo your grammar/formatting fixes!

But let's wait see what the people in charge here say about whether or not to separate it out. I don't see anything in the contributing guidelines specifying what is desired. (Maybe that's a point that would be helpful for them to clarify?)

**Comment by CBID2** [comment]
> The content change looks in line with what I was envisioning in 25763. Thank you for doing the PR.
> 
> I am not a maintainer and am new to contributing to this repo, but I believe it's generally (unless there's a repo-specific rule not to) good git practice to not combine content changes and grammar fixes in the same commit.
> 
> In this case, it made it very difficult to find the relevant change in the diff. If this content change needed to be reverted for any reason, it would then also undo your grammar/formatting fixes!
> 
> But let's wait see what the people in charge here say about whether or not to separate it out. I don't see anything in the contributing guidelines specifying what is desired. (Maybe that's a point that would be helpful for them to clarify?)

Good idea! :)

File: foundations/javascript_basics/revisiting_rock_paper_scissors.md [hunk 1]
```diff
@@ -18,7 +18,7 @@ You can make new branches by using the command `git branch <branch_name>`. You c
 
 You can see all of your current branches using `git branch` with no other arguments. The branch that you’re currently on will be indicated with an asterisk. If you want to change back to `main` from any other branch, you can do so just like changing to any other branch using `git checkout main`.
 
-Once you’re done working on your feature branch and you’re ready to bring the commits that you’ve made on it to your main branch,  you need to perform what is known as a `merge`.
+Once you’re done working on your feature branch and ready to bring the commits that you’ve made on it to your main branch,  you need to perform what is known as a `merge`.
```
Comment (rlmoser99) [inline foundations/javascript_basics/revisiting_rock_paper_scissors.md:21]: This sentence reads weird to me, in a few places. Here is a suggestion, feel free to update it. (also removed two spaces after a comma)
```suggestion
Once you are done working on your feature branch and are ready to bring the commits that you’ve made on it to your main branch, you will need to perform what is known as a `merge`.
```

File: foundations/javascript_basics/revisiting_rock_paper_scissors.md [hunk 2]
```diff
@@ -45,40 +45,40 @@ When you don’t need a branch anymore it can be deleted using `git branch -d <b
...
+* Actively learn the Git workflows discussed in this lesson with this [interactive **Visual Git Cheatsheet**](https://ndpsoftware.com/git-cheatsheet.html#loc=index;) by Andrew Peterson. **It's okay to be unfamiliar** with the variety of commands you'll interact with. You'll learn about the ones you're unfamiliar with later in the curriculum.
+* Make pushing your local commits to remote branches **easier** with the command `git push -u origin <branch>`. It automatically links the local branch you push with the remote one. [Read this educative.io article](https://www.educative.io/edpresso/what-is-the-git-push--u-remote-branch-name-command) by Talha Ashar and commit faster to a remote branch with a simple `git push` command.
+* Learn Git Branching by going through this [interactive visualizer](https://learngitbranching.js.org/) by Peter Cottle. You can learn the substantial commands about branching while seeing how the commands you write affect your visually presented branch tree.
```
Comment (rlmoser99) [inline foundations/javascript_basics/revisiting_rock_paper_scissors.md:84]: While you are updating this section, can you change the `*` to a `-`, per our style guide.

https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md#unordered-lists

**Review by rlmoser99 — CHANGES_REQUESTED** [review]
Thank you for taking the time to make these updates. I have a few changes that will need to be made.

I saw the conversation thread on the issue and want to let you know the reason that you should wait until issues have been triaged, is that we would have marked this issue as an great opportunity for a first-time contributor with our labels. We are thankful for your eagerness to contribute, but we highly value keeping easy/straightforward issues available for our students to learn how to use git and contribute back to our curriculum.

**Comment by rlmoser99** [comment]
> But let's wait see what the people in charge here say about whether or not to separate it out. I don't see anything in the contributing guidelines specifying what is desired. (Maybe that's a point that would be helpful for them to clarify?)

You're right we don't have anything in the contributing guide. I think it truly depends. Making additional changes in the same file that you are in is not too big of a deal. However, it can slow down getting things approved, as you can see, the changes that I requested are not a part of the issue.

We don't have to worry about reverting entire PRs in this repository because it is not going to breaking anything. If anything a followup PR would be used to address anything broken.

However, these are good points for code changes!

**Comment by rlmoser99** [comment]
I'm sure it is different in other repositories, but a general workflow is to let the person that requested the charges be the one to mark the conversation as resolved. That way they have the opportunity to make sure that the changes that were made actually resolve what they requested. Sometimes miscommunication happens and this helps keep the unresolved issues in the forefront of everyone's mind.

Instead, you can reply with "Done", or a link to the commit, or even a thumbs up.

File: foundations/javascript_basics/revisiting_rock_paper_scissors.md [hunk 3]
```diff
@@ -45,40 +45,41 @@ When you don’t need a branch anymore it can be deleted using `git branch -d <b
...
-*   Learn Git Branching by going through this [interactive visualizer](https://learngitbranching.js.org/) by Peter Cottle. You can learn the substantial commands about branching while seeing how the commands you write affect your visually presented branch tree.
+- Actively learn the Git workflows discussed in this lesson with this [interactive **Visual Git Cheatsheet**](https://ndpsoftware.com/git-cheatsheet.html#loc=index;) by Andrew Peterson. **It's okay to be unfamiliar** with the variety of commands you'll interact with. You'll learn about the ones you're unfamiliar with later in the curriculum.
+- Make pushing your local commits to remote branches **easier** with the command `git push -u origin <branch>`. It automatically links the local branch you push with the remote one. [Read this educative.io article](https://www.educative.io/edpresso/what-is-the-git-push--u-remote-branch-name-command) by Talha Ashar and commit faster to a remote branch with a simple `git push` command.
+
```
Comment (rlmoser99) [inline foundations/javascript_basics/revisiting_rock_paper_scissors.md:84]: Thanks for making these changes so quickly! However, this extra line will need to be removed.

Comment (CBID2) [reply foundations/javascript_basics/revisiting_rock_paper_scissors.md:84]: I'm not near my computer at the moment. Can you add your suggestion in an in-line comment,so I can commit the change please?

File: foundations/javascript_basics/revisiting_rock_paper_scissors.md [hunk 4]
```diff
@@ -45,40 +45,41 @@ When you don’t need a branch anymore it can be deleted using `git branch -d <b
...
+- Learn Git Branching by going through this [interactive visualizer](https://learngitbranching.js.org/) by Peter Cottle. You can learn the substantial commands about branching while seeing how the commands you write affect your visually presented branch tree.
```
Comment (rlmoser99) [inline foundations/javascript_basics/revisiting_rock_paper_scissors.md:85]: Sorry I didn't think about making this suggestion earlier. It does make it easier! :)
```suggestion
- Actively learn the Git workflows discussed in this lesson with this [interactive **Visual Git Cheatsheet**](https://ndpsoftware.com/git-cheatsheet.html#loc=index;) by Andrew Peterson. **It's okay to be unfamiliar** with the variety of commands you'll interact with. You'll learn about the ones you're unfamiliar with later in the curriculum.
- Make pushing your local commits to remote branches **easier** with the command `git push -u origin <branch>`. It automatically links the local branch you push with the remote one. [Read this educative.io article](https://www.educative.io/edpresso/what-is-the-git-push--u-remote-branch-name-command) by Talha Ashar and commit faster to a remote branch with a simple `git push` command.
- Learn Git Branching by going through this [interactive visualizer](https://learngitbranching.js.org/) by Peter Cottle. You can learn the substantial commands about branching while seeing how the commands you write affect your visually presented branch tree.
```

**Comment by CBID2** [comment]
Done @rlmoser99

File: foundations/javascript_basics/revisiting_rock_paper_scissors.md [hunk 5]
```diff
@@ -37,9 +37,9 @@ gitGraph
    merge develop id: "merge to main"
 </pre>
 
-Sometimes the same lines in a file will have been changed by two different branches. When this happens you will have a merge conflict when you try and merge those branches together. In order to finish merging the branches you will have to first resolve the conflict, which will be covered in a future lesson.
+Sometimes,the same lines in a file will have been changed by two different branches. When this happens, you will have a merge conflict when you try and merge those branches together. In order to finish merging the branches you will have to first resolve the conflict, which will be covered in a future lesson.
```
Comment (rlmoser99) [inline foundations/javascript_basics/revisiting_rock_paper_scissors.md:40]: This is missing a space after the first comma.
```suggestion
Sometimes, the same lines in a file will have been changed by two different branches. When this happens, you will have a merge conflict when you try and merge those branches together. In order to finish merging the branches you will have to first resolve the conflict, which will be covered in a future lesson.
```

**Review by rlmoser99 — CHANGES_REQUESTED** [review]
There is one more change, since you added more changes to the last commit.

**Comment by CBID2** [comment]
> There is one more change, since you added more changes to the last commit.

Done

**Review by rlmoser99 — APPROVED** [review]
Thank you for quickly following up on these changes!

**Comment by CBID2** [comment]
> Thank you for quickly following up on these changes!

Anytime! :)

**Pull request merged**